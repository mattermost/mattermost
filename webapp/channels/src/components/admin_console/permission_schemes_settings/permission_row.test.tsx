// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';
import {Button} from 'react-bootstrap';

import PermissionRow from 'components/admin_console/permission_schemes_settings/permission_row';

describe('components/admin_console/permission_schemes_settings/permission_row', () => {
    const defaultProps = {
        id: 'id',
        uniqId: 'uniqId',
        inherited: undefined,
        readOnly: false,
        value: 'checked',
        selectRow: jest.fn(),
        onChange: jest.fn(),
        additionalValues: {},
    };

    test('should match snapshot on editable and not inherited', () => {
        const wrapper = shallow(
            <PermissionRow {...defaultProps}/>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot on editable and inherited', () => {
        const wrapper = shallow(
            <PermissionRow
                {...defaultProps}
                inherited={{name: 'test'}}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot on read only and not inherited', () => {
        const wrapper = shallow(
            <PermissionRow
                {...defaultProps}
                readOnly={true}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot on read only and not inherited', () => {
        const wrapper = shallow(
            <PermissionRow
                {...defaultProps}
                readOnly={true}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with additional values', () => {
        const ADDITIONAL_VALUES = {
            edit_post: {
                editTimeLimitButton: (
                    <Button
                        onClick={jest.fn()}
                    />
                ),
            },
        };

        const wrapper = shallow(
            <PermissionRow
                {...defaultProps}
                additionalValues={ADDITIONAL_VALUES}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should call onChange function on click', () => {
        const onChange = jest.fn();
        const wrapper = shallow(
            <PermissionRow
                {...defaultProps}
                onChange={onChange}
            />,
        );
        wrapper.find('div').first().simulate('click');
        expect(onChange).toBeCalledWith('id');
    });

    test('shouldn\'t call onChange function on click when is read-only', () => {
        const onChange = jest.fn();
        const wrapper = shallow(
            <PermissionRow
                {...defaultProps}
                readOnly={true}
                onChange={onChange}
            />,
        );
        wrapper.find('div').first().simulate('click');
        expect(onChange).not.toBeCalled();
    });
});
