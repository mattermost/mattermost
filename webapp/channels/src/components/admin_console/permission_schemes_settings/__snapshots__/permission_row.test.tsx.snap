// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/admin_console/permission_schemes_settings/permission_row should match snapshot on editable and inherited 1`] = `
<div
  className="permission-row"
  id="uniqId"
  onClick={[Function]}
>
  <PermissionCheckbox
    id="uniqId-checkbox"
    value="checked"
  />
  <span
    className="permission-name"
  >
    id
  </span>
  <PermissionDescription
    additionalValues={Object {}}
    description=""
    id="id"
    inherited={
      Object {
        "name": "test",
      }
    }
    selectRow={[MockFunction]}
  />
</div>
`;

exports[`components/admin_console/permission_schemes_settings/permission_row should match snapshot on editable and not inherited 1`] = `
<div
  className="permission-row"
  id="uniqId"
  onClick={[Function]}
>
  <PermissionCheckbox
    id="uniqId-checkbox"
    value="checked"
  />
  <span
    className="permission-name"
  >
    id
  </span>
  <PermissionDescription
    additionalValues={Object {}}
    description=""
    id="id"
    selectRow={[MockFunction]}
  />
</div>
`;

exports[`components/admin_console/permission_schemes_settings/permission_row should match snapshot on read only and not inherited 1`] = `
<div
  className="permission-row read-only"
  id="uniqId"
  onClick={[Function]}
>
  <PermissionCheckbox
    id="uniqId-checkbox"
    value="checked"
  />
  <span
    className="permission-name"
  >
    id
  </span>
  <PermissionDescription
    additionalValues={Object {}}
    description=""
    id="id"
    selectRow={[MockFunction]}
  />
</div>
`;

exports[`components/admin_console/permission_schemes_settings/permission_row should match snapshot on read only and not inherited 2`] = `
<div
  className="permission-row read-only"
  id="uniqId"
  onClick={[Function]}
>
  <PermissionCheckbox
    id="uniqId-checkbox"
    value="checked"
  />
  <span
    className="permission-name"
  >
    id
  </span>
  <PermissionDescription
    additionalValues={Object {}}
    description=""
    id="id"
    selectRow={[MockFunction]}
  />
</div>
`;

exports[`components/admin_console/permission_schemes_settings/permission_row should match snapshot with additional values 1`] = `
<div
  className="permission-row"
  id="uniqId"
  onClick={[Function]}
>
  <PermissionCheckbox
    id="uniqId-checkbox"
    value="checked"
  />
  <span
    className="permission-name"
  >
    id
  </span>
  <PermissionDescription
    additionalValues={
      Object {
        "edit_post": Object {
          "editTimeLimitButton": <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            onClick={[MockFunction]}
          />,
        },
      }
    }
    description=""
    id="id"
    selectRow={[MockFunction]}
  />
</div>
`;
